system/system_monitor/src/process_monitor/process_monitor.cpp

```diff
@@ -50,34 +54,23 @@ ProcessMonitor::ProcessMonitor(const rclcpp::NodeOptions & options)
     memory_tasks_.push_back(task);
     updater_.add(*task);
   }
-}
 
-void ProcessMonitor::update() { updater_.force_update(); }
+  // Start timer to execute top command
+  timer_ = rclcpp::create_timer(this, get_clock(), 1s, std::bind(&ProcessMonitor::onTimer, this));
```

Comment:
Since the onTimer function and diagnostics_updater work in the same callback group, separate the callback groups.
```suggestion
  timer_callback_group_ = this->create_callback_group(rclcpp::CallbackGroupType::MutuallyExclusive);
  auto sub_opt = rclcpp::SubscriptionOptions();
  sub_opt.callback_group = timer_callback_group_;
  timer_ = rclcpp::create_timer(this, get_clock(), 1s, std::bind(&ProcessMonitor::onTimer, this), sub_opt);

  // rclcpp::CallbackGroup::SharedPtr timer_callback_group_; /* defined as member variables in this class */
```

Reply:
@yn-mrse Thank you for the great suggestion. Fixed in https://github.com/autowarefoundation/autoware.universe/pull/948/commits/d2151975376fcaadd100b590db9ba5bf4ae0ef98.